1-Draft/RFCNNNN-Native-Markdown-Rendering.md

```diff
@@ -0,0 +1,141 @@
+---
+RFC: 
+Author: Aditya Patwardhan
+Status: Draft
+SupercededBy: 
+Version: 0.1
+Area: Formatting and Output
+Comments Due: 03/26/2018
+Plan to implement: Yes
+---
+
+# Native Markdown Rendering
+
+Render markdown content on console to improve readability. 
+
+## Motivation
+
+Markdown is a common authoring format used by the community.
+There is no easy way in PowerShell to visualize a markdown document on console.
+
+## Specification
+
+This RFC proposes to use VT100 escape sequences to render markdown content.
+`ConvertFrom-Markdown` cmdlet as part of `Microsoft.PowerShell.MarkdownRender` PowerShell module would consume a string or a file path and output a VT100 encoded string.
+Optionally, the output can be formatted as HTML.
+For converting strings to VT100 coded strings, we will be writing an extension to [markdig](https://github.com/lunet-io/markdig).
+The extension will insert VT100 escape sequences as appropriate.
+
+```PowerShell
+ConvertFrom-Markdown [-Path] <string[]> -AsHTML -AsPlainText  [<CommonParameters>]
+
+ConvertFrom-Markdown -LiteralPath <string[]> -AsHTML -AsPlainText  [<CommonParameters>]
+
+ConvertFrom-Markdown -InputObject <psobject> -AsHTML -AsPlainText  [<CommonParameters>]
+```
+
+### Parameters
+
+- **Path** : Accepts an array of file paths with markdown content.
+- **LiteralPath** : Accepts an array of literal paths with markdown content.
+- **InputObject** : Accepts an InputObject of `System.IO.FileSystemInfo`, `string` type.
+- **AsHTML** : When selected, the output will be HTML.
+- **AsPlainText** : When selected, the output will be plain text.
+
+The default output will be a string encoded with VT100 escape sequences.
```

Comment:
This cmdlet is not intended solely for the console output. So I guess users can expect another default (more populat HTML).

Reply:
I prefer suggestion from @DerpMcDerp, I will update the RFC to state that, `ConvertFrom-Markdown` returns an object, and the default formatter renders it as markdown.

Reply:
I agree that we could return an object. And replace `AsPlainText`  with `AsVT100Encoded`.
If we use simplified markdown scheme why we need property `Tokens`? It seems useless.
Having two results (Html and VT100EncodedString) at once is too costly. I suggest use methods or one property `Value`.

Reply:
> replace AsPlainText with AsVT100Encoded

I agree.

> why we need property Tokens

I think we will get the parsed tokens/Asts anyway during the conversion. @DerpMcDerp can you give an example where the parsed markdown ASTs may be useful?

> Having two results (Html and VT100EncodedString) at once is too costly

We don't generate two results in one go. By default, only `Html` is populated; if `-AsVT100Encoded` is specified, then `VT100EncodedString` is populated. This is similar to `MeasureInfo` objects produced from `Measure-Object`.